client/src/main/java/org/apache/celeborn/client/write/PushState.java

```diff
@@ -88,4 +96,20 @@ public boolean addBatchData(String addressPair, PartitionLocation loc, int batch
   public DataBatches takeDataBatches(String addressPair) {
     return batchesMap.remove(addressPair);
   }
+
+  public void addFlightBatches(int batchId, PartitionLocation loc) {
+    String addressPair = loc.hostAndPushPort();
+    Set<Integer> batchIdSetPerPair =
+        batchIdPerAddressPair.computeIfAbsent(addressPair, id -> new HashSet<>());
+    batchIdSetPerPair.add(batchId);
+  }
+
+  public void removeFlightBatches(int batchId, PartitionLocation loc) {
```

Comment:
There can be concurrent calls to this method, but batchIdSetPerPair is not thread-safe.

Reply:
hello, @waitinfuture